lmfit/model.py

```diff
@@ -1027,22 +1027,23 @@ def fit(self, data, params=None, weights=None, method='leastsq',
             if not np.isscalar(kwargs[var]):
                 kwargs[var] = _align(kwargs[var], mask, data)
 
-        # Make sure `dtype` for data is always `float64` or `complex128`
-        if np.isrealobj(data):
-            data = np.asfarray(data)
-        elif np.iscomplexobj(data):
-            data = np.asarray(data, dtype='complex128')
-
-        # Coerce `dtype` for independent variable(s) to `float64` or
-        # `complex128` when the variable has one of the following types: list,
-        # tuple, numpy.ndarray, or pandas.Series
+        def coerce_arraylike(x):
+            """
+            coerce lists, tuples, and pandas Series to float64 or complex128,
+            but leave other ndarrays of different dtypes and objects unchanged
+            """
+            if isinstance(x, (list, tuple, Series)):
+                if np.isrealobj(x):
+                    return np.asfarray(x)
+                elif np.iscomplexobj(x):
+                    return np.asarray(x, dtype='complex128')
+            return x
+
+        # coerce array-like (list, tuples, Series) data and
+        # independent variable(s) to float64/complex128
```

Comment:
is this comment completely accurate? It seems to me from the code that independent variables will only be coerced if the are lists/tuples/pandas.Series. The way I read the comment is that data will be coerced if belonging to the categories listed and that independent variable(s) will always be coerced.

Reply:
will try to clarify the comment.